### PR TITLE
Fix state diff issue

### PR DESCRIPTION
**Pull request by aasoni**

### CODE REVIEW

File: Scripts/SpacetimeDBClient.cs [hunk 1]
```diff
@@ -528,67 +523,48 @@ struct ProcessedMessage
...
             {
-                var message = preProcessedMessage.message;
-                var timestamp = preProcessedMessage.timestamp;
-                var dbOps = preProcessedMessage.dbOps;
-                // Perform the state diff, this has to be done on the main thread because we have to touch
-                // the client cache.
-                if (message.TypeCase == Message.TypeOneofCase.SubscriptionUpdate)
+                foreach (var table in clientDB.GetTables())
                 {
-                    foreach (var table in clientDB.GetTables())
```
Comment (jdetter) [inline Scripts/SpacetimeDBClient.cs:562]: Yeah this was a major issue - basically you can have updates that are in the queue which are "in front" of the update you're looking at and then you calculate an incorrect state diff. 

TLDR its good that we're moving this to the main thread
